docs/fsharp/language-reference/slices.md

```diff
@@ -0,0 +1,136 @@
+let unboundedEnd = fullList.[94..]
+printfn "Unbounded end slice: %A" unboundedEnd
+```
+
+Slicing arrays is also identical:
+
+```fsharp
+// Generate an array of 100 integers
+let fullArray = [| 1 .. 100 |]
+
+// Create a slice from indices 1-5 (inclusive)
+let smallSlice = fullArray.[1..5]
+printfn "Small slice: %A" smallSlice
+
+// Create a slice from the beginning to index 5 (inclusive)
+let unboundedBeginning = fullArray.[..5]
+printfn "Unbounded beginning slice: %A" unboundedBeginning
+
+// Create a slice from an index to the end of the list
+let unboundedEnd = fullArray.[94..]
+printfn "Unbounded end slice: %A" unboundedEnd
+```
+
+## Slicing multidimensional arrays
+
+F# supports multidimensional arrays in the F# core library. As with one-dimensional arrays, they can be useful to slices of. However, the introduction of additional dimensions mandates a slightly different syntax so that you can take slices of specific rows and columns.
+
+The following examples demonstrate how to slice a 2D array:
+
+```fsharp
+// Generate a 3x3 2D matrix
+let A = array2D [[1;2;3];[4;5;6];[7;8;9]]
+printfn "Full matrix:\n %A" A
+
+// Take the first row
+let row0 = A.[0,*]
+printfn "Row 0: %A" row0
+
+// Take the first column
+let col0 = A.[*,0]
+printfn "Column 0: %A" col0
+
+// Take all rows but only two columns
+let subA = A.[*,0..1]
+printfn "%A" subA
+
+// Take two rows and all columns
+let subA' = A.[0..1,*]
+printfn "%A" subA'
+
+// Slice a 2x2 matrix out of the full 3x3 matrix
+let twoByTwo = A.[0..1,0..1]
+printfn "%A" twobyTwo
+```
+
+The F# core library does not define `GetSlice`for 3D arrays. If you wish to slice those or other arrays of higher dimensions, you must define the `GetSlice` member yourself.
+
+## Defining slices for other data structures
+
+The F# core library defines slices for a limited set of types. If you wish to define slices for more data types, you can do so either in the type definition itself or in a type extension.
+
+For example, here's how you might define slices and and indexed property for the `ArraySegment` class to allow for convenient data manipulation:
+
+```fsharp
+open System
+
+type ArraySegment<'TItem> with
+    member segment.Item(index) =
+        segment.Array.[segment.Offset + index]
+
+    member segment.GetSlice(?start, ?finish) =
+        let start = defaultArg start 0
+        let finish = defaultArg finish segment.Count
+        ArraySegment(segment.Array, segment.Offset + start, finish - start)
+
+let arr = ArraySegment [| 1 .. 10 |]
+let slice = arr.[2..5] //[ 3; 4; 5]
+```
+
+### Use inlining to avoid boxing if it is necessary
+
+If you are defining slices for a type that is actually a struct, it is recommend that you `inline` the `GetSlice` member. The F# compiler will optimize away the optional arguments, avoiding any heap allocations as a result of slicing. This is critically important for slicing constructs such as `Span<T>` that cannot be be allocated on the heap.
```

Comment:
- it is recommend --> we recommend
- will optimize --> optimizes
- Do you want a link to Span\<T>? \<xref:System.Span`1>.